javascript/preflight.js

```diff
@@ -0,0 +1,46 @@
+function runPreFlightChecks() {
+  const isHttp      = window.location.protocol == 'http:',
+        isLocalhost = window.location.hostname == 'localhost'
+        isUnsafe    = !(isHttp && isLocalhost);
+
+  if (isUnsafe) {
+    const isHttpMessage = [
+      'This page is running over the web over HTTP.',
+      'You should use HTTPS and rotate your access token.',
+    ].join(' ');
+    alert(isHttpMessage);
+  }
+
+  // We will:
+  // - Make a request to the github rate limit endpoint
+  //   (This will not affect the rate limit)
+  // - Check what scopes we have access to
+  const token             = new FourthWall.getQueryVariables().token,
+        ghUrl             = 'https://api.github.com/rate_limit',
+        authGhUrl         = ghUrl + '?access_token=' + token;
+
+  fetch(authGhUrl)
+    .then(function (response) { return response.headers; })
+    .then(function (headers)  { return headers.get('x-oauth-scopes')  })
+    .then(function (scopes)   { return scopes.split(', '); })
+    .then(function (scopes)   {
+      const allowedScopes = ['repo:status', 'repo_deployment'];
```

Comment:
We need to add `read:org` to this list as it's used to get repos by team membership (mentioned in the README):
https://github.com/alphagov/fourth-wall#how-to-use

I'd also suggest updating the README to be more clear about which scopes are necessary and which (read:org) are optional. 

After that I'd be happy to merge.